### ISSUE TITLE
Multiple AtlasDB connections

### ISSUE THREAD

**Issue by david-apple-labdoor** [paralect/robomongo#1291]
I have three atlas db instances and have all 3 configured as sets in robomongo.  I can connect to any one of them on the Robomongo's first launch but I can only connect to one at a time.  If I try to connect to a second, I get an error that "sets primary is unreachable".  All three have the same set name which may be causing the issue.

**Comment by simsekgokhan** [comment]
Hi @david-apple-labdoor , we have two clusters on Mongo Atlas, and we are able to connect and work on both of them without problems.

Can you help us to reproduce your problem? My understanding is you have one atlas account and have 3 clusters, each cluster has 3 nodes. (9 nodes in total) But in this case, set name is unique and given automatically by atlas.

Our Config:
```
Cluster0: 5 nodes
Members: cluster0-shard-00-0x-xxxx.mongodb.net:27017
Set Name: Cluster0-shard-0

Cluster1: 3 nodes
Members: cluster1-shard-00-0x-xxxx.mongodb.net:27017
Set Name: Cluster1-shard-0
```

**Comment by david-apple-labdoor** [comment]
The issue we have is they have the same setname, IE:
3 node cluster:  cluster-shard-00-0x-xxxx.mongodb.net:2701
3 node cluster:  cluster-shard-00-0x-yyyy.mongodb.net:2701
3 node cluster:  cluster-shard-00-0x-zzzz.mongodb.net:2701

All of these end up with the set name set as "cluster-shard-0".  

They are effectively identical databases for different environments.  It may be that the solution is to rebuild them under new unique names.

**Comment by simsekgokhan** [comment]
Thanks for the info. But I am still not sure about same replica set name. Because Mongo Atlas gives set name unique every time you create a new cluster. 

Maybe, you have same set name because you have 3 different Atlas accounts, do you?
We have one Atlas account.

Can you also share your URI connection strings for each of your set?

**For example our's is below:**
`mongodb://admin:<PASSWORD>@cluster0-shard-00-00-xyx.mongodb.net:27017,cluster0-shard-00-01-xyz.mongodb.net:27017,cluster0-shard-00-02-xyz.mongodb.net:27017/<DATABASE>?ssl=true&replicaSet=Cluster0-shard-0&authSource=admin`

Here, the set name is `Cluster0-shard-0` which is `replicaSet` parameter's value.

And, yes I agree with you that recreating sets with different set names is worth to try as a solution.

**Comment by david-apple-labdoor** [comment]
Only 1 account.  Here are the connection strings which should illustrate the issue.  Both of ours are the same except for the 5 letters before .mongodb.net.  This is in one account but those identifiers are the only difference.  

mongodb://admin:<PASSWORD>@services-shard-00-00-xxxxx.mongodb.net:27017,services-shard-00-01-xxxxx.mongodb.net:27017,services-shard-00-02-xxxxx.mongodb.net:27017/<DATABASE>?ssl=true&replicaSet=services-shard-0&authSource=admin

and

mongodb://admin:<PASSWORD>@services-shard-00-00-yyyyy.mongodb.net:27017,services-shard-00-01-yyyyy.mongodb.net:27017,services-shard-00-02-yyyyy.mongodb.net:27017/<DATABASE>?ssl=true&replicaSet=services-shard-0&authSource=admin

And because of this, both replicaset parameters end up being "services-shard-0" in robomongo.json which I'm guessing is at least part of the issue.

**Comment by simsekgokhan** [comment]
Yes, seems like you have same set names. Then, we will try to replicate your config to reproduce this problem. Thanks for responding with the requested information again.

**Comment by simsekgokhan** [comment]
And can you help us maybe telling us how you were able to create two sets with same name?
We have tried but Mongo Atlas website does not accept same cluster name while creating a new cluster.

We would like to reproduce the problem for investigation but currently we are unable to reproduce.

**Comment by david-apple-labdoor** [comment]
Let me check in with the individual that created the sets.  I've taken over
the admin of them but wasn't around for their creation.  I will find out
and let you know.  Thank you for your assistance in helping with this.

**Comment by simsekgokhan** [comment]
Ok, thanks a lot, we have reproduced the problem finally.
Definitely not a simple issue at first glance, actually more than a problem, seems like design intent. 
I was able to connect to replica sets with same name only with opening two Robomongo instances.

The root cause of this issue is little deep inside the design of MongoDB. Robomongo uses MongoDB code and the code (i.e. mongo.exe) is designed to connect to a replica set at a time. (i.e. users need to open two mongo.exe to connect to two replica sets). So connecting to two replica sets with same set name from a "single" mongo.exe is not supported by Mongo, that's what Robomongo tries to do and fails.

**Solution 1:**
---
The quickest solution is to open a Robomongo instance for each replica set that has "the same set name". 

**Solution 2:**
---
Second solution is just to change set name (cluster name) in Mongo Atlas or add a new cluster with a different name.

Please let us know if any of the solution works.
Best Regards.

**Comment by juliashibalko** [comment]
Hi @david-apple-labdoor! Could you please check the issue with the latest Robo3T 1.1? As in Robo3T 1.0 we have included fix for ReplicaSet name and you can change it manually in connection settings. Also it will show error in case you have same repset names. Refer to http://blog.robomongo.org/robomongo-1-0/
Please share with us how it went.

**Comment by delino12** [comment]
Navigate to setting added your system IP address on the white-list section. Solved.
